neuralpp/symbolic/expression.py

```diff
@@ -346,6 +353,23 @@ def __str__(self) -> str:
         return f"{self.function}({argument_str})"
 
 
+class Context(Expression, ABC):
```

Comment:
Added a new subclass of `Expression`: `Context`. It has a `dict` property to provide variable-value lookup, and a `unsatisfiable` property if the condition is unsatisfiable.